<launch> 
   <!-- Stereo Odometry -->   
   <!-- <node pkg="tf" type="static_transform_publisher" name="camera_to_base" args="0 0 0 -1.57079632679 0 0 base_link camera_infra1_optical_frame 100"/> -->

   <node pkg="rtabmap_ros" type="rgbd_odometry" name="rgbd_odometry" output="screen">
      <remap from="rgb/image"       to="/camera/infra1/image_rect_raw"/>
      <remap from="rgb/camera_info" to="/camera/infra1/camera_info"/>
      <remap from="depth/image"     to="/camera/depth/image_rect_raw"/>
      <remap from="odom"            to="vis_odom"/>
      
      <param name="queue_size" type="int" value="2"/>
      
      <param name="Odom/Strategy" type="string" value="1"/> <!-- 0=Frame to map, 1=frame to frame -->
      <param name="Vis/CorType" value="1"/>
      <param name="Vis/EstimationType" type="string" value="0"/> <!-- 3D->2D (PnP) -->
      <param name="Vis/MinInliers" type="string" value="10"/>
      <param name="Vis/RoiRatios" type="string" value="0.03 0.03 0.04 0.04"/>
      <param name="Vis/MaxDepth" type="string" value="10"/>
      <param name="Vis/CorNNDR" type="string" value="0.8"/>
      <param name="Vis/MaxFeatures" type="string" value="500"/>
      <param name="Vis/FeatureType" type="string" value="8"/>
      <param name="wait_for_transform_duration" type="string" value="0.5"/> 
      <param name="Odom/KeyFrameThr" type="string" value= "0"/>
      <param name="Odom/FilteringStrategy" type="string" value= "0"/>
      <param name="Vis/BundleAdjustment" type="int" value="0"/>
      <param name="Odom/KalmanMeasurementNoise" type="string" value="0.3"/>
      <param name="Vis/ForwardEstOnly" type="string" value="true"/>
      <param name="Odom/ResetCountdown" type="int" value="1"/>
      <param name="frame_id" type="string" value="camera_link"/>
      <param name="odom_frame_id" type="string" value="map"/>
      <param name="wait_for_transform_duration" type="string" value="0.5"/> 
   </node>

</launch>
